File: src/main/java/com/enhe/endpoint/util/NavigateUtil.java
package com.enhe.endpoint.util;

import com.intellij.codeInsight.navigation.NavigationUtil;
import com.intellij.psi.PsiElement;
import com.intellij.ui.awt.RelativePoint;
import com.intellij.util.PsiNavigateUtil;

import java.awt.event.MouseEvent;
import java.util.List;

/**
 * @author ding.shichen
 */
public class NavigateUtil {

    public static void openTargetsMethod(List<PsiElement> elements, String title, MouseEvent e) {
        if (elements.size() == 1) {
            PsiNavigateUtil.navigate(elements.get(0));
        } else {
            NavigationUtil.getPsiElementPopup(elements.toArray(PsiElement[]::new), title).show(new RelativePoint(e));
        }
    }
}

File: src/main/kotlin/com/enhe/endpoint/util/NavigateUtil.kt
package com.enhe.endpoint.util

import com.intellij.codeInsight.navigation.getPsiElementPopup
import com.intellij.psi.PsiElement
import com.intellij.ui.awt.RelativePoint
import com.intellij.util.PsiNavigateUtil
import java.awt.event.MouseEvent;

object NavigateUtil {

    @JvmStatic
    fun openTargetsMethod(elements: List<PsiElement>, title: String, e: MouseEvent) {
        if (elements.size == 1) {
            PsiNavigateUtil.navigate(elements.first());
        } else {
            getPsiElementPopup(elements.toTypedArray(), title).show(RelativePoint(e))
        }
    }
}